lms/static/scripts/frontend_apps/utils/test/api-test.js

```diff
@@ -672,6 +672,20 @@ describe('usePaginatedAPIFetch', () => {
     assert.equal(getHasMorePages(wrapper), 'Yes');
   });
 
+  it('does not reload hasMorePages while loading', () => {
```

Comment:
```suggestion
  it('does not reset `hasMorePages` while loading', () => {
```

Or alternatively "preserves `hasMorePages` while loading next page".